src/natcap/invest/sdr/sdr_core.pyx

```diff
@@ -557,4 +567,141 @@ def calculate_sediment_deposition(
                         global_col, global_row, r_i)
                     f_raster.set(global_col, global_row, f_i)
 
+    LOGGER.info('100% complete')
     sediment_deposition_raster.close()
+
+
+def calculate_average_flow(
+    mfd_flow_direction_path, target_average_flow_path):
+    """Calculate the Average Proportional Flow from MFD.
+
+    Calculates the weighted average of proportional flow for every MFD flow
+    direction pixel provided.
+
+    Parameters:
+        mfd_flow_direction_path (string): The path to an MFD flow direction
+            raster.
+        target_average_flow_path (string): The path to where the calculated
+            average proportional flow raster should be written.
+
+    Returns:
+        ``None``.
+
+    """
+    LOGGER.info('Calculating average aspect')
+
+    cdef float average_flow_nodata = -1.0
+    pygeoprocessing.new_raster_from_base(
+        mfd_flow_direction_path, target_average_flow_path,
+        gdal.GDT_Float32, [average_flow_nodata])
+
+    flow_direction_info = pygeoprocessing.get_raster_info(
+        mfd_flow_direction_path)
+    cdef float mfd_flow_direction_nodata = flow_direction_info['nodata'][0]
+    cdef int n_cols, n_rows
+    n_cols, n_rows = flow_direction_info['raster_size']
+
+    cdef _ManagedRaster mfd_flow_direction_raster = _ManagedRaster(
+        mfd_flow_direction_path, 1, False)
+
+    cdef _ManagedRaster average_flow_raster = _ManagedRaster(
+        target_average_flow_path, 1, True)
+
+    cdef int* neighbor_flows = [0, 0, 0, 0, 0, 0, 0, 0]
```

Comment:
These aren't "flows" so much as "weights"... In other places in this module "flow" refers to the flux that's flowing through the pixel. Could we change the nomenclature here too to be consistent? Like weights if it's those 8 bit values from the 64 bit and then "aspect" when calculating the sin+cos values?